### ISSUE TITLE
[Bug]: Anthropic set api key error

### ISSUE THREAD

**Issue by haodong-cc** [infiniflow/ragflow#3263]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

7e0148c

### Actual behavior

A small bug caused the inability to add anthropic model providers.
I found that this bug is caused by this code:
![20241107171049](https://github.com/user-attachments/assets/41709abd-de19-4507-b1e0-21b6d3b2dbd4)

The file is located at /rag/llm/chat_model.py  row1261
This should not be parsed as Json string, causing the following code to fail in taking values:
```ans = response["content"][0]["text"]```

Suggestion for modification:
change .json() to .dict()

### Steps to reproduce

```Markdown
when i add anthropic model providers and set apikey, always fail.
```

**Comment by KevinHuSh** [comment]
Could you paste the error message or `print(response)` if it's convenient ?

**Comment by haodong-cc** [comment]
> Could you paste the error message or `print(response)` if it's convenient ?

Yes, my negligence. This was an error in setting the anthropic’s apikey.
![20241108101151](https://github.com/user-attachments/assets/9c0ec1ef-4fb0-4502-a6da-e111a79e3323)

Additionally, I have ignored claude-instant-1.2 as it caused an error:
 ```Fail to access model(claude-instant-1.2) using this api key. **ERROR**: Error code: 404 - {'type': 'error', 'error': {'type': 'not_found_error', 'message': 'model: claude-instant-1.2'}```

**Comment by kevinroleke** [comment]
Same issue

```
 Fail to access model(claude-2.0) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-2.1) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-5-sonnet-20240620) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-haiku-20240307) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-opus-20240229) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-sonnet-20240229) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-instant-1.2) using this api key. **ERROR**: Error code: 404 - {'type': 'error', 'error': {'type': 'not_found_error', 'message': 'model: claude-instant-1.2'}}
 ```

**Comment by mattsahn** [comment]
> Same issue
> 
> ```
>  Fail to access model(claude-2.0) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-2.1) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-5-sonnet-20240620) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-haiku-20240307) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-opus-20240229) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-3-sonnet-20240229) using this api key. **ERROR**: string indices must be integers, not 'str' Fail to access model(claude-instant-1.2) using this api key. **ERROR**: Error code: 404 - {'type': 'error', 'error': {'type': 'not_found_error', 'message': 'model: claude-instant-1.2'}}
> ```

Same error for me right now.

**Comment by yuzhichang** [comment]
Fixed by #3351
